### PR TITLE
89 navigation styling - Micheale Andemariam

### LINKED ISSUE #89: Navigation styling
## Acceptance Criteria

Add navigation user dropdown (when logged in)
Add initial only - look up charAt() method, at index 0
image

Style navigation links and add underline functionality for current page

## Implementation Notes
as above

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by micheale2021**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #89

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Comment by micheale2021** [comment]
I have added a hover effect to NavLinks but have not added underline functionality for the current page. I have the code/logic but it's taking some time to implement it with the existing code. If we have enough time I will create a ticket and implement that.
Thank you.

File: client/src/components/NavBar.js [hunk 1]
```diff
@@ -3,57 +3,67 @@ import { Container, Nav, Navbar, Image } from "react-bootstrap";
...
+                 height="50"
+               />
+             </Navbar.Brand>
+             <Navbar.Toggle aria-controls="basic-navbar-nav" />
+             <Navbar.Collapse
+               className="d-flex justify-content-end"
+               id="basic-navbar-nav"
+             >
+               <Nav className="p-2">
+                 <Nav.Link className="nav-link active" href="/all-users">
```
Comment (RainbowGiantSquid) [inline client/src/components/NavBar.js]: good idea of using the class, however active class should be applied dynamically based on current page displayed. so if user is on 'all users' the that menu item has the active class

File: client/src/components/main.css [hunk 2]
```diff
@@ -19,4 +19,24 @@ body {
   margin: 50px auto;
   text-align: center;
   color: #ff0000;
-}
\ No newline at end of file
+}
+
```
Comment (RainbowGiantSquid) [inline client/src/components/main.css]: navigation link text is not align in the centre, but the hover works nice, should be a slightly different colour though.

image

Comment (RainbowGiantSquid) [reply client/src/components/main.css]: Also the dropdown button with the initals needs styling to be round or at least the chevron displayed on the same line height
image

**Review by RainbowGiantSquid — CHANGES_REQUESTED** [review]
A couple of comments to address, but other than that looking great!

File: client/src/components/NavBar.js (hunk 1 above)
Comment (micheale2021) [reply client/src/components/NavBar.js]: For now, I've removed the class name active. I will be working on and implementing it on a separate ticket.

File: client/src/components/main.css (hunk 2 above)
Comment (micheale2021) [reply client/src/components/main.css]: Chevron is displayed on the same line, the hover effect is removed for now.

**Comment by micheale2021** [comment]
Let's merge this, for now, it has a nav bar background colour light. I will work on a separate ticket for the rest.

**Review by RainbowGiantSquid — APPROVED** [review]
Looking good - I've resolved conflicts and removed the 'active' - perhaps the adding 'active' is a candidate for the further implementation - this class should be added dynamically when the current page path equals the link. so if you are on /all-users, the 'active' class is added to the 'All Users' link.
Anyway, all good.

**Pull request merged**